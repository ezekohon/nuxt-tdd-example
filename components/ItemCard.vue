<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-danger">
          {{ itemData.title }}
        </p>
      </header>
      <footer class="card-footer" :class="footerClass" @click="toggleBuyed($vnode.key)">
        <p class="card-footer-item is-justify-content-flex-end has-text-weight-bold">
          {{ `$${itemData.price}` }}
        </p>
      </footer>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  props: {
    itemData: {
      type: Object,
      required: true
    }
  },
  computed: {
    footerClass () {
      return this.itemData.buyed ? 'card-footer--buyed' : 'card-footer--not-buyed'
    }
  },
  methods: {
    ...mapActions('items', ['toggleBuyed'])
    // toggleBuyed () {
    //   // this.itemData.buyed = !this.itemData.buyed
    //   this.toggleBuyed(this.$vnode.key)
    // }
  }
}
</script>

<style lang="scss" scoped>
.card-footer {
  cursor: pointer;
}
.card-footer--buyed {
  background-color: greenyellow;
}
.card-footer--not-buyed {
  background-color: orangered;
}
.is-justify-content-flex-end {
  justify-content: flex-end;
}
</style>
