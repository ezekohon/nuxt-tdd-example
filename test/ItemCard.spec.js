import { mount } from '@vue/test-utils';
import ItemCard from '@/components/ItemCard'

let wrapper = null;

const props = {
    itemData: {
        title: 'title',
        price: 21,
        buyed: false
    }
};

beforeEach(() => {
    wrapper = mount(ItemCard, {
        propsData: {
            itemData: props.itemData
        }
    });
})

afterEach(() => {
    wrapper.destroy();
})

describe('ItemCard.vue', () => {

    it('renders title ', () => {
        expect(wrapper.find(".card-header-title").text()).toBe(props.itemData.title);
    })

    it('renders price ', () => {
        expect(wrapper.find(".card-footer-item").text()).toContain(props.itemData.price);
    })

    it('has red background if not buyed', () => {

        expect(wrapper.find(".card-footer").classes()).toContain('card-footer--not-buyed');
    })

    it('has green background if buyed', () => {
        const props = {
            itemData: {
                title: 'title',
                price: 21,
                buyed: true
            }
        };

        wrapper = mount(ItemCard, {
            propsData: {
                itemData: props.itemData
            }
        });

        expect(wrapper.find(".card-footer").classes()).toContain('card-footer--buyed');
    })

    it('matches snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
})