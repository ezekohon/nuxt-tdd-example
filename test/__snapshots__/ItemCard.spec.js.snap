// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard.vue matches snapshot 1`] = `
<div
  class="column"
>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title has-text-danger"
      >
        
        title
      
      </p>
    </header>
     
    <footer
      class="card-footer card-footer--not-buyed"
    >
      <p
        class="card-footer-item is-justify-content-flex-end has-text-weight-bold"
      >
        
        $21
      
      </p>
    </footer>
  </div>
</div>
`;
